### PR TITLE
Multi-gpu compatibility; Optimized tensor loading on GPU 

### PR DESCRIPTION
**Pull request by garyfanhku**
Wrapped model with DataParallel when multiple gpus are detected; load input tensor on GPU only when GPUs are visible in case CPU inference is to be performed.

### CODE REVIEW

**Comment by garyfanhku** [comment]
A few remarks:
1. As is discussed in #10 , multi-GPU training performance might not be guaranteed. You can also follow the process of Synced BN development in PR #11   
2. To follow the linear scaling rule in https://arxiv.org/abs/1706.02677, you have to manually adjust the learning rate and batch-size accordingly.
